buildSrc/src/main/groovy/bio.terra.landingzone.java-common-conventions.gradle

```diff
@@ -42,23 +42,14 @@ dependencies {
     compileOnly "com.google.code.findbugs:annotations:3.0.1"
     implementation 'org.slf4j:slf4j-api'
     implementation 'io.swagger.core.v3:swagger-annotations:2.1.12'
+    implementation 'io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:2.2.0'
     swaggerCodegen 'io.swagger.codegen.v3:swagger-codegen-cli:3.0.47'
 
     testImplementation 'ch.qos.logback:logback-classic'
     testImplementation 'org.hamcrest:hamcrest:2.2'
 
-    var openTelemetryVersion = '1.32.0'
-    implementation "io.opentelemetry:opentelemetry-api:${openTelemetryVersion}"
-    implementation "io.opentelemetry:opentelemetry-sdk:${openTelemetryVersion}"
-    implementation "io.opentelemetry:opentelemetry-sdk-metrics:${openTelemetryVersion}"
-    implementation "io.opentelemetry:opentelemetry-exporter-logging:${openTelemetryVersion}"
-    implementation "io.opentelemetry.semconv:opentelemetry-semconv:1.21.0-alpha"
-    implementation "io.opentelemetry.instrumentation:opentelemetry-spring-webmvc-6.0:${openTelemetryVersion}-alpha"
-    implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:${openTelemetryVersion}"
-    implementation "io.opentelemetry.instrumentation:opentelemetry-spring-boot:${openTelemetryVersion}-alpha"
-    implementation "com.google.cloud.opentelemetry:exporter-trace:0.25.2"
-
-    implementation ('bio.terra:terra-common-lib:0.1.6-SNAPSHOT'){
+    implementation 'com.google.guava:guava:33.1.0-jre' // Needed for TCL
```

Comment:
If TCL needs guava, why doesn't it include it?

Reply:
Thanks for asking.  My theory (and I wish I were more solid on this) is that TCL exposes guava as an `api` dependency but does not explicitly version it, because TCL derives its version from Google's libraries-bom:

https://github.com/DataBiosphere/terra-common-lib/blame/698ea95e495af35810cabf631a238e5a4a91384b/build.gradle#L74

But LZS does not pull in Google's libraries-bom (TCL does not expose it as an `api` dependency) so it can't find guava:

<img width="1594" alt="Screenshot 2024-03-22 at 11 18 33 AM" src="https://github.com/DataBiosphere/terra-landing-zone-service/assets/79769153/9f44295c-a3d7-4bb8-a508-5fdaf695e639">

I am wondering if TCL should instead expose Google's libraries-bom as an `api` dependency, or if not, if LZS should instead pull in Google's libraries-bom itself.

This has been a question that's come up in other ways for me recently (exposing an `api` dependency whose version is derived from a BOM without exposing the BOM itself) so I'm going to raise it to a wider audience in #dsp-engineering.

Reply:
https://broadinstitute.slack.com/archives/C0DSD41QT/p1711121985316039

Reply:
For now, I'm inclined to pull in Google's libraries-bom instead with a more expansive comment.  Hopefully we can change this upstream in TCL but I want to discuss that strategy more widely.

Reply:
Flip-flopped a bit: I updated the comment, but I ended up keeping the direct versioned guava dependency for now.  We talked in stand-up about excluding it when pulling in TCL, but turns out that LZS does use it after all, such as for `@VisibleForTesting` annotations.

Longer term, we (the wider we) will need to consciously decide what we want TCL to be, and how opinionated it should be about dependency definition and exposure to consumers.